python/lsst/analysis/drp/gatherResourceUsage.py

```diff
@@ -0,0 +1,550 @@
+# This file is part of analysis_drp.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+__all__ = (
+    "GatherResourceUsageConfig",
+    "GatherResourceUsageConnections",
+    "GatherResourceUsageTask",
+)
+
+import argparse
+from collections import defaultdict
+import dataclasses
+import fnmatch
+import re
+import logging
+
+import numpy as np
+import pandas as pd
+
+from lsst.daf.butler import Butler, DataCoordinate, DatasetRef, DatasetType, Quantum
+from lsst.pex.config import Field, ListField
+from lsst.pipe.base import (
+    PipelineTask,
+    PipelineTaskConfig,
+    PipelineTaskConnections,
+    QuantumGraph,
+    Struct,
+    TaskDef,
+)
+from lsst.pipe.base import connectionTypes as ct
+from lsst.utils.introspection import get_full_type_name
+
+# It's not great to be importing a private symbol, but this is a temporary
+# workaround for the fact that prior to w.2022.10, the units for memory values
+# written in task metadata were platform-dependent.  Once we no longer care
+# about older runs, this import and the code that uses it can be removed.
+from lsst.utils.usage import _RUSAGE_MEMORY_MULTIPLIER
+
+
+_LOG = logging.getLogger(__name__)
+
+
+class GatherResourceUsageConnections(PipelineTaskConnections, dimensions=()):
+    output_table = ct.Output(
+        "PLACEHOLDER_resource_statistics",  # Should always be overridden.
+        storageClass="DataFrame",
+        dimensions=(),
+        doc=(
+            "Table that aggregates memory and CPU usage statistics from one "
+            "or more tasks. "
+            "This will have one row for each data ID, with columns for each "
+            "task or method's memory usage and runtime."
+        ),
+    )
+    input_metadata = ct.Input(
+        "PLACEHOLDER_metadata",  # Should always be overridden.
```

Comment:
`PLACEHOLDER` in both places seems to imply these are linked together. If that is true, would using a template make sense here?

Reply:
Good idea; I always forget about the templates.